linera-base/src/identifiers.rs

```diff
@@ -169,22 +169,15 @@ pub enum BlobType {
 
 impl Display for BlobType {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            BlobType::Data => write!(f, "Data")?,
-        };
-
-        Ok(())
+        write!(f, "{}", serde_json::to_string(self).unwrap())
     }
 }
 
 impl FromStr for BlobType {
     type Err = anyhow::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "Data" => Ok(BlobType::Data),
-            _ => Err(anyhow!("Invalid blob type: {}", s)),
-        }
+        serde_json::from_str(s).map_err(|_| anyhow!("Invalid BlobType: {}", s))
```

Comment:
```suggestion
        serde_json::from_str(s).with_context(|| format!("Invalid BlobType: {}", s))
```

That way the original error is included in the `anyhow` error as well.

Reply:
Thx! Done https://github.com/linera-io/linera-protocol/pull/2364/commits/cfd2bd1e13b91677abb6b162831b39f0a3d35aa6